.generator/src/generator/templates/api.j2

```diff
@@ -88,9 +81,131 @@ func (a *{{ classname }}Service) {{ operation.operationId }}(ctx _context.Contex
 	}
 	{%- endif %}
 	{%- endfor %}
+	return req, nil
+}
+
+/*
+ * {{ operation.operationId }} {{ operation.summary }}
+{{ operation.description|block_comment(prefix=" *") }}
+ */
+func (a *{{ classname }}Service) {{ operation.operationId }}(ctx _context.Context{% for name, parameter in operation|parameters if parameter.required or parameter.in == "path" %}, {{ name|variable_name}} {{ get_type_for_parameter(parameter) }}{% endfor %}{%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}{%- if loop.first %}, o ...{{ operation.operationId }}OptionalParameters{% endif %}{% endfor %}) ({% if returnType %}{{ returnType }}, {% endif %}*_nethttp.Response, error) {
+	req, err := a.build{{ operation.operationId }}Request(ctx{% for name, parameter in operation|parameters if parameter.required or parameter.in == "path" %}, {{ name|variable_name}}{% endfor %}{%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}{%- if loop.first %}, o...{% endif %}{% endfor %})
+	if err != nil {
+		{%- if returnType %}
+		var localVarReturnValue {{ returnType }}
+		{%- endif %}
+		return {% if returnType %}localVarReturnValue, {% endif %}nil, err
+	}
 
 	return req.ApiService.{{ operation.operationId|untitle_case }}Execute(req)
 }
+{%- if operation["x-pagination"] %}
+{%- set pagination = operation["x-pagination"] %}
+/*
+ * {{ operation.operationId }}WithPagination provides a paginated version of {{ operation.operationId }} returning a channel with all items.
+ */
+{%- set itemType = get_type_at_path(operation, pagination.resultsPath) %}
+func (a *{{ classname }}Service) {{ operation.operationId }}WithPagination(ctx _context.Context{% for name, parameter in operation|parameters if parameter.required or parameter.in == "path" %}, {{ name|variable_name}} {{ get_type_for_parameter(parameter) }}{% endfor %}{%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}{%- if loop.first %}, o ...{{ operation.operationId }}OptionalParameters{% endif %}{% endfor %}) (items chan {{ itemType }}, cancel func(), err error) { 
```

Comment:
```suggestion
func (a *{{ classname }}Service) {{ operation.operationId }}WithPagination(ctx _context.Context{% for name, parameter in operation|parameters if parameter.required or parameter.in == "path" %}, {{ name|variable_name}} {{ get_type_for_parameter(parameter) }}{% endfor %}{%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}{%- if loop.first %}, o ...{{ operation.operationId }}OptionalParameters{% endif %}{% endfor %}) (items <-chan {{ itemType }}, cancel func(), err error) { 
```
Might be better to return a read-only channel.

Reply:
I had to remove parameter names.